src/components/ui/header.tsx

```diff
@@ -262,7 +263,7 @@ const Header = ({ type, pageTitle, sharehouseId }: IHeaderContainerProps) => {
         ) : type === 'HeaderItemWithDropDown' ? (
           <HeaderItemWithDropDown
             pageTitle={pageTitle}
-            sharehouseId={sharehouseId!}
+            sharehouseId={sharehouseId ?? ''}
```

Comment:
Ideally, `sharehouseId` should **never be undefined** when the `type` is `HeaderItemWithDropdown`. I understand the issue mentioned in the PR description, so when addressing it, please ensure `sharehouseId` is always defined for `HeaderItemWithDropdown` type.